src/mlpack/core/tree/binary_space_tree/binary_space_tree_impl.hpp

```diff
@@ -1083,25 +1083,25 @@ void BinarySpaceTree<MetricType, StatisticType, MatType, BoundType, SplitType>::
     right = NULL;
   }
 
-  ar & BOOST_SERIALIZATION_NVP(begin);
-  ar & BOOST_SERIALIZATION_NVP(count);
-  ar & BOOST_SERIALIZATION_NVP(bound);
-  ar & BOOST_SERIALIZATION_NVP(stat);
+  ar & CEREAL_NVP(begin);
+  ar & CEREAL_NVP(count);
+  ar & CEREAL_NVP(bound);
+  ar & CEREAL_NVP(stat);
 
-  ar & BOOST_SERIALIZATION_NVP(parentDistance);
-  ar & BOOST_SERIALIZATION_NVP(furthestDescendantDistance);
-  ar & BOOST_SERIALIZATION_NVP(dataset);
+  ar & CEREAL_NVP(parentDistance);
+  ar & CEREAL_NVP(furthestDescendantDistance);
+  ar & CEREAL_NVP(dataset);
 
-  // Save children last; otherwise boost::serialization gets confused.
+  // Save children last; otherwise cereal gets confused.
   bool hasLeft = (left != NULL);
   bool hasRight = (right != NULL);
 
-  ar & BOOST_SERIALIZATION_NVP(hasLeft);
-  ar & BOOST_SERIALIZATION_NVP(hasRight);
+  ar & CEREAL_NVP(hasLeft);
+  ar & CEREAL_NVP(hasRight);
   if (hasLeft)
-    ar & BOOST_SERIALIZATION_NVP(left);
+    ar & CEREAL_NVP(left);
   if (hasRight)
-    ar & BOOST_SERIALIZATION_NVP(right);
+    ar & CEREAL_NVP(right);
```

Comment:
Here's an idea for serializing the children that avoids serializing a pointer directly:

```
  bool hasLeftChild = (this->left != NULL);
  bool hasRightChild = (this->right != NULL);
  ar & BOOST_SERIALIZATION_NVP(hasLeftChild);
  ar & BOOST_SERIALIZATION_NVP(hasRightChild);

  // at this point I know if I have a left or right child
  if (Archive::is_loading::value)
  {
    if (hasLeftChild)
    {
      delete this->left;
      left = new TreeType(); // allocate memory for it, so that *this->left is
                             // valid when we deserialize into it
    }

    if (hasRightChild)
    {
      delete this->right;
      right = new TreeType(); // allocate memory for it, so that *this->right is
                              // valid when we deserialize into it
    }
  }

  if (this->left != NULL)
    ar & BOOST_SERIALIZATION_NVP(*this->left);

  if (this->right != NULL)
    ar & BOOST_SERIALIZATION_NVP(*this->right);
```

I think that will work but I'm not completely sure.